orchagent/fgnhgorch.cpp

```diff
@@ -0,0 +1,1379 @@
+            bank_member_changes[fgNhgEntry->nextHops[nhk.ip_address]].
+                nhs_to_add.push_back(nhk);
+            next_hop_to_add = true;
+        }
+        else 
+        {
+            FGNextHopGroupEntry *syncd_fg_route_entry = &(syncd_fg_route_entry_it->second);
+            if(syncd_fg_route_entry->active_nexthops.find(nhk) == 
+                syncd_fg_route_entry->active_nexthops.end())
+            {
+                bank_member_changes[fgNhgEntry->nextHops[nhk.ip_address]].
+                    nhs_to_add.push_back(nhk);
+            }
+        }
+
+        sai_object_id_t next_hop_id = m_neighOrch->getNextHopId(nhk);
+        nhopgroup_members_set[nhk] = next_hop_id;
+    }
+
+    if(syncd_fg_route_entry_it != m_syncdFGRouteTables.at(vrf_id).end())
+    {
+        FGNextHopGroupEntry *syncd_fg_route_entry = &(syncd_fg_route_entry_it->second);
+
+        /* Route exists, update FG ECMP group in SAI */
+        for(auto nhk : syncd_fg_route_entry->active_nexthops)
+        {
+            if(nhopgroup_members_set.find(nhk) == nhopgroup_members_set.end())
+            {
+                bank_member_changes[fgNhgEntry->nextHops[nhk.ip_address]].
+                    nhs_to_del.push_back(nhk);
+            }
+            else
+            {
+                bank_member_changes[fgNhgEntry->nextHops[nhk.ip_address]].
+                    active_nhs.push_back(nhk);
+            }
+        }
+
+        if(!compute_and_set_hash_bucket_changes(syncd_fg_route_entry, fgNhgEntry, bank_member_changes, 
+                nhopgroup_members_set, ipPrefix))
+        {
+            return false;
+        }
+    }
+    else
+    {
+        /* New route + nhg addition */
+        if(next_hop_to_add == false)
+        {
+            SWSS_LOG_INFO("There were no valid next-hops to add %s:%s", ipPrefix.to_string().c_str(),
+                    nextHops.to_string().c_str());
+            /* Let the route retry logic(upon false rc) take care of this case */
+            return false;
+        }
+
+        if(gRouteOrch->getNextHopGroupCount() >= gRouteOrch->getMaxNextHopGroupCount())
+        {
+            SWSS_LOG_DEBUG("Failed to create new next hop group. \
+                        Reached maximum number of next hop groups.");
+            return false;
+        }
+
+        FGNextHopGroupEntry syncd_fg_route_entry;
+        string platform = getenv("platform") ? getenv("platform") : "";
+
+        sai_attribute_t nhg_attr;
+        vector<sai_attribute_t> nhg_attrs;
+
+        nhg_attr.id = SAI_NEXT_HOP_GROUP_ATTR_TYPE;
+        nhg_attr.value.s32 = SAI_NEXT_HOP_GROUP_TYPE_FINE_GRAIN_ECMP;
+        nhg_attrs.push_back(nhg_attr);
+
+        nhg_attr.id = SAI_NEXT_HOP_GROUP_ATTR_CONFIGURED_SIZE;
+        nhg_attr.value.s32 = fgNhgEntry->configured_bucket_size;
+        nhg_attrs.push_back(nhg_attr);
+
+        sai_object_id_t next_hop_group_id;
+        sai_status_t status = sai_next_hop_group_api->create_next_hop_group(&next_hop_group_id,
```

Comment:
Please move this section of calling sai to a different function. This is already a big function.